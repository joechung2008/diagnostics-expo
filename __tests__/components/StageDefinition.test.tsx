import StageDefinition from '@/components/StageDefinition';
import { render } from '@testing-library/react-native';

// Mock the ThemeContext
jest.mock('@/components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      tableBorder: '#cccccc',
      tableHeaderBg: '#e0e0e0',
    },
  }),
}));

describe('StageDefinition', () => {
  const mockStageDefinition = {
    development: ['dev-server-1', 'dev-server-2'],
    production: ['prod-server-1'],
    staging: ['staging-server-1', 'staging-server-2', 'staging-server-3'],
  };

  it('renders stage definition title', () => {
    const { getByText, toJSON } = render(
      <StageDefinition stageDefinition={mockStageDefinition} />
    );

    expect(getByText('Stage Definitions')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders all stage keys and values correctly', () => {
    const { getByText, toJSON } = render(
      <StageDefinition stageDefinition={mockStageDefinition} />
    );

    expect(getByText('Key')).toBeTruthy();
    expect(getByText('Value')).toBeTruthy();

    // Check development stage
    expect(getByText('development')).toBeTruthy();
    expect(getByText('dev-server-1, dev-server-2')).toBeTruthy();

    // Check production stage
    expect(getByText('production')).toBeTruthy();
    expect(getByText('prod-server-1')).toBeTruthy();

    // Check staging stage
    expect(getByText('staging')).toBeTruthy();
    expect(
      getByText('staging-server-1, staging-server-2, staging-server-3')
    ).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders single stage definition', () => {
    const singleStage = { production: ['prod-server-1'] };
    const { getByText, toJSON } = render(
      <StageDefinition stageDefinition={singleStage} />
    );

    expect(getByText('production')).toBeTruthy();
    expect(getByText('prod-server-1')).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders empty stage definition', () => {
    const emptyStage = {};
    const { getByText, toJSON } = render(
      <StageDefinition stageDefinition={emptyStage} />
    );

    expect(getByText('Stage Definitions')).toBeTruthy();
    expect(getByText('Key')).toBeTruthy();
    expect(getByText('Value')).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders stage with empty server list', () => {
    const stageWithEmptyList = { empty: [] };
    const { getByText, toJSON } = render(
      <StageDefinition stageDefinition={stageWithEmptyList} />
    );

    expect(getByText('empty')).toBeTruthy();
    expect(getByText('')).toBeTruthy(); // Empty string for empty array

    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with complete stage definition data', () => {
    const { toJSON } = render(
      <StageDefinition stageDefinition={mockStageDefinition} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
