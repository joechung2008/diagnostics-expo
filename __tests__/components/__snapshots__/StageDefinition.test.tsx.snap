// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageDefinition matches snapshot with complete stage definition data 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        development
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        dev-server-1, dev-server-2
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        production
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        prod-server-1
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging-server-1, staging-server-2, staging-server-3
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders all stage keys and values correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        development
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        dev-server-1, dev-server-2
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        production
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        prod-server-1
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging-server-1, staging-server-2, staging-server-3
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders empty stage definition 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders single stage definition 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        production
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        prod-server-1
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders stage definition title 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        development
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        dev-server-1, dev-server-2
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        production
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        prod-server-1
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        staging-server-1, staging-server-2, staging-server-3
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders stage with empty server list 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        empty
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
